src/modules/renderPage.js

```diff
@@ -1,67 +1,103 @@
-import { Link } from '../helpers.js';
+import { Link, updateHistoryRecord } from '../helpers.js';
+import { query, queryAll, nextTick, compareArrays } from '../utils.js';
 
-const renderPage = function(page, popstate) {
+const renderPage = function(page, { popstate, fragment } = {}) {
 	document.documentElement.classList.remove('is-leaving');
 
 	const isCurrentPage = this.getCurrentUrl() === page.url;
-	if (!isCurrentPage) return;
+	if (!isCurrentPage) {
+		console.log('not current page:', this.getCurrentUrl(), 'vs', page.url, page);
+		return;
+	}
+
+	const skipTransition = fragment || (popstate && !this.options.animateHistoryBrowsing);
```

Comment:
I've seen this in a couple of places. Maybe also a good candidate for util function?

Reply:
I have a rule to extract util functions if a line is repeated in 3 places or more. Is there a 3rd one I missed? I was counting 2.